cadc-vos-server/src/main/java/ca/nrc/cadc/vos/server/transfers/TransferUtil.java

```diff
@@ -229,6 +230,51 @@ else if ("https".equalsIgnoreCase(scheme))
         return protocolList;
     }
 
+    /**
+     * Generate a Protocol list with endpoints set to /vault/pkg/<jobid>. One entry for every protocol in
+     * the transfer provided. Use the jobID from the job provided.
+     * @param transfer
+     * @param job
+     * @return
+     * @throws MalformedURLException
+     * @throws MalformedURLException
+     */
+    public static List<Protocol> getPackageEndpoints(final Transfer transfer, final Job job)
+        throws MalformedURLException, IllegalArgumentException {
+
+        // package view is redirected to /vault/pkg/<jobid>
+        // making an endpoint for each protocol provided
+
+        List<Protocol> protocolList = transfer.getProtocols();
+
+        if (!protocolList.isEmpty()) {
+
+            StringBuilder sb = new StringBuilder();
+            String jobID = job.getID();
+            sb.append("/").append(jobID);
+
+            RegistryClient regClient = new RegistryClient();
+            LocalServiceURI localServiceURI = new LocalServiceURI();
+            URI serviceURI = localServiceURI.getURI();
+            AuthMethod authMethod = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
+            URL serviceURL = regClient.getServiceURL(serviceURI, Standards.PKG_10, authMethod);
+            if (serviceURL == null) {
+                throw new RuntimeException("Couldn't get serviceURL for URI: " + serviceURI.toString());
+            }
+            URL location = new URL(serviceURL.toExternalForm() + sb.toString());
+            String loc = location.toExternalForm();
+            log.debug("Package endpoint: " + loc);
+
+            for (Protocol p : protocolList) {
+                p.setEndpoint(loc);
```

Comment:
There is a function in Capabilities to get an individual Capability by Standard - used that. For a Capability, there is a function to get interfaces by auth method. Opted to use this because if I iterated through peach interface in each protocol, I'd have to compare the authMethod anyway. 
The function will return only the protocols that endpoints exist for. (ie, if VOS.PROTOCOL_HTTP_GET is requested, no protocol for it ends up in the transfer doc returned because it doesn't exist in the capabilities doc.)